### PR TITLE
Update load functions

### PR DESCRIPTION
**Pull request by jaclynbeck-sage**
Previously, `load.py` had `"./staging"` hard-coded in several places as the folder where staged json files go before upload to Synapse. This became an issue when trying to write test functions for our various custom transforms, because files in ./staging would get over-written with test data unless I wrote some really awful patch code to get around it. 

To fix this, I added `- staging_path: ./staging` to the config files, and whatever is in `staging_path` now gets passed as an argument to the load functions. By default, if `- staging_path` is missing from the config file, it will get set to "./staging" so it behaves as before. 

I also added a utility function, `_find_config_by_name`, to get the correct config data from the config list. Previously, objects were retrieved from the config list by numerical index (e.g. `datasets = config[1]["datasets"]`), which meant that either `staging_path` had to get lost at the bottom of the config file or I had to change the indexes on each reference to config. Adding a function to access by name instead of numerical index seemed more robust. 

Looking for input from Data & Tooling about these changes!

### CODE REVIEW

File: agoradatatools/etl/load.py [hunk 1]
```diff
@@ -149,11 +149,11 @@ def df_to_csv(df: pd.core.frame.DataFrame, filename: str):
     return temp_csv.name
 
 
-def dict_to_json(df: dict, filename: str):
+def dict_to_json(df: dict, staging_path: str, filename: str):
     try:
         df_as_dict = [{d: remove_non_values(v) if isinstance(v, dict) else v for d,v in df.items()}]
 
-        temp_json = open("./staging/" + filename, 'w+')
+        temp_json = open(staging_path + "/" + filename, 'w+')
```
Comment (thomasyu888) [inline agoradatatools/etl/load.py]: Nit: using os.path.join will be cleaner

File: agoradatatools/etl/load.py [hunk 2]
```diff
@@ -17,21 +17,21 @@ def default(self, obj):
         return json.JSONEncoder.default(self, obj)
 
 
-def create_temp_location():
+def create_temp_location(staging_path: str):
```
Comment (thomasyu888) [inline agoradatatools/etl/load.py:20]: I did see this and I had planned on @BWMac to eventually use the tempfile module in python to create these temp locations.

That said, thanks for contributing this refactor.  This is an excellent improvement from what was being done.

File: agoradatatools/etl/utils.py [hunk 3]
```diff
@@ -30,3 +30,28 @@ def _get_config(config_path: str = None):
...
         sys.exit(errno.EBADF)
     return config
+
+
+def _find_config_by_name(config: list, name: str):
+    """
+    Iterates through the list to find a dictionary with a key matching 'name'
+
+    Parameters
+    ----------
```
Comment (thomasyu888) [inline agoradatatools/etl/utils.py]: Nit: we use trying to change to using Google style docs to be consistent across the entire package. We will slowly make this change in the coming year.

```
Args:
     ....

Returns:
    ....
```

File: agoradatatools/process.py [hunk 4]
```diff
@@ -107,26 +109,36 @@ def process_all_files(config_path: str = None, syn=None):
...
+    
     if datasets:
         for dataset in datasets:
             new_syn_tuple = process_dataset(
-                dataset_obj=dataset, syn=syn
+                dataset_obj=dataset, staging_path=staging_path, syn=syn
             )
             # in the future we should log new_syn_tuples that are none
 
+    destination = utils._find_config_by_name(config, "destination")
```
Comment (thomasyu888) [inline agoradatatools/process.py:128]: There are some key redactors that I want to do with the config that will basically enable us to not have to use this function.

TLDR, @JessterB knows which Jira ticket it is, but it is the one to add yaml anchors for all the synapse ids in the config.

Great idea for the meantime!

**Review by thomasyu888 — COMMENTED** [review]
LGTM, @BWMac can you add comments?

I'd be happy to give a more thorough read through when I'm back from vacay if necessary.

File: agoradatatools/etl/load.py [hunk 5]
```diff
@@ -103,7 +103,7 @@ def load(file_path: str, provenance: list[str], destination: str, syn=None):
     return (file.id, file.versionNumber)
 
 
-def df_to_json(df: pd.core.frame.DataFrame, filename: str):
+def df_to_json(df: pd.core.frame.DataFrame, staging_path: str, filename: str):
```
Comment (BWMac) [inline agoradatatools/etl/load.py]: NIT: you can get away with just having `pd.DataFrame` for your typing

**Review by BWMac — APPROVED** [review]
Not much to comment on, solid PR. For the docstrings, if you are already using vscode and the [autoDocstring](https://marketplace.visualstudio.com/items?itemName=njpwerner.autodocstring) extension, you can just change your `  autoDocstring.docstringFormat` variable in `settings.json` to "google" to automatically have the format @thomasyu888 mentioned.

File: agoradatatools/etl/load.py (hunk 1 above)
Comment (jaclynbeck-sage) [reply agoradatatools/etl/load.py]: Good catch, fixed in latest commit.

File: agoradatatools/etl/utils.py (hunk 3 above)
Comment (jaclynbeck-sage) [reply agoradatatools/etl/utils.py]: Fixed in latest commit.

File: agoradatatools/etl/load.py (hunk 5 above)
Comment (jaclynbeck-sage) [reply agoradatatools/etl/load.py]: Fixed in latest commit.

**Review by thomasyu888 — APPROVED** [review]

**Comment by thomasyu888** [comment]
@jaclynbeck-sage Feel free to merge!
